src/test/coverartcache_test.cpp

```diff
@@ -18,43 +18,73 @@ class CoverArtCacheTest : public MixxxTest, public CoverArtCache {
     virtual void TearDown() {
     }
 
+    void loadCoverFromMetadata(QString trackLocation) {
+        CoverInfo info;
+        info.type = CoverInfo::METADATA;
+        info.source = CoverInfo::GUESSED;
+        info.coverLocation = QString();
+        info.trackLocation = trackLocation;
+
+        CoverArtCache::FutureResult res;
+        res = CoverArtCache::loadCover(info, NULL, 1234, 0, false);
+        EXPECT_EQ(1234, res.requestReference);
+        EXPECT_QSTRING_EQ(QString(), res.cover.info.coverLocation);
+        EXPECT_QSTRING_EQ(info.hash, res.cover.info.hash);
+
+        SecurityTokenPointer securityToken = Sandbox::openSecurityToken(
+            QDir(trackLocation), true);
+        SoundSourceProxy proxy(trackLocation, securityToken);
+        Mixxx::SoundSourcePointer pSoundSource(proxy.getSoundSource());
+        ASSERT_TRUE(pSoundSource);
+        QImage img = pSoundSource->parseCoverArt();
+        
+        EXPECT_FALSE(img.isNull());
+        EXPECT_EQ(img, res.cover.image);
```

Comment:
Not here. This is a coverArt CACHE test. we want to test the cache. so it is sufficient to test if 
1 - the image from the cache is identical to the image directly extracted from track 
2 - they are not null (this is what was missing)

The right place to do it is in the CoverArtUtil::extractEmbeddedCover test. If it is ok there, it will be ok here.